code/game/objects/items/weapons/caution.dm

```diff
@@ -24,7 +24,7 @@
 /obj/item/caution/proximity_sign/attack_self(mob/user as mob)
 	if(ishuman(user))
 		var/mob/living/carbon/human/H = user
-		if(H.mind.assigned_role != "Janitor")
+		if(!(H.mind.has_antag_datum(/datum/antagonist/traitor) || H.mind.has_antag_datum(/datum/antagonist/mindslave)))
```

Comment:
I'm not sure if you want ONLY mindslaves here and not vampire thralls as well, but if you do, please use `ismindlsave()`. The difference is that this helper proc has `ignore_subtypes = TRUE` on the `has_antag_datum` so it bars `/datum/antagonist/mindslave/thrall`.

Reply:
Sorry for taking so long to respond, I've been out of town.
For whatever reason, use of the proc resulted in mindslaves being unable to activate or deactivate the signs, so I've implemented a hacky method below to bar them from use. If you have any suggestions on how to make it start behaving, just let me know!